ethereum/deploy_util.go

```diff
@@ -81,28 +75,73 @@ func DeployGSNForwarderWithKey(
 		return ethcommon.Address{}, err
 	}
 
-	err = registerDomainSeparator(ctx, ec, privkey, address, contract)
+	err = registerDomainSeparator(ctx, ec, privKey, address, contract)
 	if err != nil {
 		return ethcommon.Address{}, err
 	}
 
 	return address, nil
 }
 
+func isDomainSeparatorRegistered(
+	ctx context.Context,
+	ec *ethclient.Client,
+	forwarderAddr ethcommon.Address,
+	forwarder *gsnforwarder.Forwarder,
+) (bool, error) {
+	chainID, err := ec.ChainID(ctx)
+	if err != nil {
+		return false, err
+	}
+	name := gsnforwarder.DefaultName
+	version := gsnforwarder.DefaultVersion
+	ds, err := common.GetEIP712DomainSeparator(name, version, chainID, forwarderAddr)
+	if err != nil {
+		return false, err
+	}
+	return forwarder.Domains(nil, ds) // isRegistered, error
```

Comment:
same the return values instead of the comment?

also nil `*bind.CallOpts` is okay? did not know that

Reply:
It's the same as passing an empty `&bind.CallOpts{}`.  If you are calling a view-only function that, for some reason, accesses `msg.sender`, it won't work.  Looking at the structure, I should have passed it for the context field.  I'll change it:
```
type CallOpts struct {
	Pending     bool            // Whether to operate on the pending state or the last known one
	From        common.Address  // Optional the sender address, otherwise the first account is used
	BlockNumber *big.Int        // Optional the block number on which the call should be performed
	Context     context.Context // Network context to support cancellation and timeouts (nil = no timeout)
}
```